src/test/scala/cognite/spark/v1/FlexibleDataModelConnectionRelationTest.scala

```diff
@@ -0,0 +1,198 @@
+import com.cognite.sdk.scala.v1.fdm.instances.{InstanceCreate, SlimNodeOrEdge}
+import com.cognite.sdk.scala.v1.fdm.views.ViewReference
+import org.apache.spark.sql.{DataFrame, Row}
+import org.scalatest.{FlatSpec, Matchers}
+
+import scala.concurrent.duration.DurationInt
+import scala.util.{Success, Try}
+
+class FlexibleDataModelConnectionRelationTest
+    extends FlatSpec
+    with Matchers
+    with SparkTest
+    with FlexibleDataModelsTestBase {
+
+  private val startEndNodeContainerExternalId = "sparkDsConnectionsTestContainerStartEndNodes1"
+  private val startEndNodeViewExternalId = "sparkDsConnectionsTestViewStartEndNodes1"
+  private val propsMap = Map(
+    "stringProp1" -> FDMContainerPropertyTypes.TextPropertyNonListWithoutDefaultValueNullable
+  )
+  private val connectionsViewExtId = "sparkDsTestConnectionsView"
+
+  //  client.spacesv3.createItems(Seq(SpaceCreateDefinition(spaceExternalId))).unsafeRunSync()
+
+  it should "fetch all connection instances" in {
+    val edgeTypeExtId = s"edgeTypeExternalId${apiCompatibleRandomString()}"
+    val startNodeExtId = s"${startEndNodeViewExternalId}FetchStartNode"
+    val endNodeExtId = s"${startEndNodeViewExternalId}FetchEndNode"
+
+    val results = (for {
+      startEndNodeContainer <- createContainerIfNotExists(
+        usage = Usage.Node,
+        propsMap,
+        startEndNodeContainerExternalId
+      )
+      startEndNodeView <- createViewWithCorePropsIfNotExists(
+        startEndNodeContainer,
+        startEndNodeViewExternalId,
+        viewVersion)
+      _ <- createNodesForEdgesIfNotExists(
+        startNodeExtId,
+        endNodeExtId,
+        startEndNodeView.toSourceReference,
+      )
+
+      connectionsContainer <- createContainerIfNotExists(
+        usage = Usage.Edge,
+        propsMap,
+        "connectionSourceContainer")
+      connectionSourceView <- createViewWithCorePropsIfNotExists(
+        connectionsContainer,
+        "connectionSourceView",
+        viewVersion)
+      _ <- createViewWithConnectionsIfNotExists(
+        connectionSourceView.toSourceReference,
+        `type` = DirectRelationReference(spaceExternalId, externalId = edgeTypeExtId),
+        connectionsViewExtId,
+        viewVersion
+      )
+      connectionsWritten <- createConnectionWriteInstances(
+        typeNode = DirectRelationReference(space = spaceExternalId, externalId = edgeTypeExtId),
+        startNode = DirectRelationReference(space = spaceExternalId, externalId = startNodeExtId),
+        endNode = DirectRelationReference(space = spaceExternalId, externalId = endNodeExtId)
+      )
+      _ <- IO.sleep(2.seconds)
+    } yield connectionsWritten).unsafeRunSync()
+
+    val readConnectionsDf = readRows(edgeSpace = spaceExternalId, edgeExternalId = edgeTypeExtId)
+
+    readConnectionsDf.createTempView("connection_instances_table")
+
+    val selectedConnectionInstances = spark
+      .sql("select * from connection_instances_table")
```

Comment:
I think we need to ingest different edge"types" so that we can have a test:
```sql
SELECT * FROM connection_instances_table WHERE type = named_struct"space", $typeSpace, "externalId", $typeExteralId) AND space = $edgeSpace
```
to make sure the filter work on different edge types and spaces 😁